controller/program/tests/tests/remove_lst.rs

```diff
@@ -0,0 +1,615 @@
+        .with_lst_token_program(*token_program)
+        .build()
+}
+
+fn remove_lst_ix(keys: &RemoveLstIxKeysOwned, lst_idx: u32) -> Instruction {
+    let accounts = keys_signer_writable_to_metas(
+        keys.0.iter(),
+        REMOVE_LST_IX_IS_SIGNER.0.iter(),
+        REMOVE_LST_IX_IS_WRITER.0.iter(),
+    );
+    Instruction {
+        program_id: Pubkey::new_from_array(ID),
+        accounts,
+        data: RemoveLstIxData::new(lst_idx).as_buf().into(),
+    }
+}
+
+fn remove_lst_fixtures_accounts_opt(keys: &RemoveLstIxKeysOwned) -> Vec<PkAccountTup> {
+    fixtures_accounts_opt_cloned(keys.0.iter().copied()).collect()
+}
+
+fn assert_correct_remove(bef: &[PkAccountTup], aft: &[PkAccountTup], mint: &[u8; 32]) {
+    let lst_state_lists = acc_bef_aft(&Pubkey::new_from_array(LST_STATE_LIST_ID), bef, aft);
+    let [_, lst_state_list_acc_aft] = lst_state_lists;
+
+    let [lst_state_list_bef, lst_state_list_aft]: [Vec<_>; 2] =
+        lst_state_lists.each_ref().map(|a| {
+            LstStatePackedList::of_acc_data(&a.data)
+                .unwrap()
+                .0
+                .iter()
+                .map(|x| x.into_lst_state())
+                .collect()
+        });
+
+    let bef_len = lst_state_list_bef.len();
+
+    if bef_len == 1 {
+        assert!(lst_state_list_acc_aft.data.is_empty() && lst_state_list_acc_aft.lamports == 0);
+    } else {
+        let diffs = LstStateListChanges::new(&lst_state_list_bef)
+            .with_del_by_mint(mint)
+            .build();
+
+        assert_diffs_lst_state_list(&diffs, &lst_state_list_bef, &lst_state_list_aft);
+    }
+}
+
+#[test]
+fn remove_lst_jupsol_fixture() {
+    let pool_pk = Pubkey::new_from_array(POOL_STATE_ID);
+    let pool_acc = ALL_FIXTURES
+        .get(&pool_pk)
+        .expect("missing pool state fixture");
+    let pool = PoolStatePacked::of_acc_data(&pool_acc.data)
+        .unwrap()
+        .into_pool_state();
+
+    let admin = pool.admin;
+    let token_program = &TOKENKEG_ID;
+    let refund_rent_to = Pubkey::new_unique().to_bytes();
+
+    // Find jupSOL in the list to get its index
+    let lst_state_list_acc = ALL_FIXTURES
+        .get(&Pubkey::new_from_array(LST_STATE_LIST_ID))
+        .expect("missing lst state list fixture");
+    let lst_state_list = LstStatePackedList::of_acc_data(&lst_state_list_acc.data)
+        .unwrap()
+        .0;
+
+    let jupsol_idx = lst_state_list
+        .iter()
+        .position(|packed| {
+            let lst = unsafe { packed.as_lst_state() };
+            lst.mint == JUPSOL_MINT.to_bytes()
+        })
+        .expect("jupSOL not found in fixture list");
+
+    let mut lst_state_list_data = lst_state_list_acc.data.clone();
+    let lst_state_list = LstStatePackedListMut::of_acc_data(&mut lst_state_list_data).unwrap();
+    let lst_state = lst_state_list.0.get_mut(jupsol_idx).unwrap();
+    // safety: account data is 8-byte aligned
+    unsafe { lst_state.as_lst_state_mut().sol_value = 0 };
```

Comment:
so this isnt always true offchain but i think i fukt up and didnt provide mutator methods for `LstStatePacked`. This may be UB if your OS decides to give you a block of memory thats not 8-byte aligned. Probably safe but inefficient way for now would be to convert to `Vec<LstState>`, modify, then convert back to bytes.